### ISSUE TITLE
Calling find_tables() on one document alters the bounding boxes of a subsequent document

### ISSUE THREAD

**Issue by richardpaulhudson** [pymupdf/PyMuPDF#3001]
### Description of the bug

If I open one PDF and call `find_tables()` on a page that has a table, and then open a completely different PDF, the bounding boxes returned for the second PDF are different from when I perform the same sequence of operations but commenting out the `find_tables()` line. Reloading the `fitz` module does not solve the problem. 

### How to reproduce the bug

I have added the example documents `with_table.pdf` and `without_table.pdf` in the first comment below. The same behaviour appears to occur with many, but not all, PDFs with tables.

```
import fitz

with_table_filename = ".../with_table.pdf"
without_table_filename = ".../without_table.pdf"

with_tables_doc = fitz.open(with_table_filename)
with_tables_doc[1].find_tables() 

without_table_doc = fitz.open(without_table_filename)
blocks = without_table_doc[0].get_text("blocks")
print(blocks[0][:4])
```

Running the code above as-is outputs:

```
(56.79999923706055, 58.24467468261719, 221.9661865234375, 70.24467468261719)
```

but running it with the `find_tables()` line commented out outputs:

```
(56.79999923706055, 57.2073974609375, 221.9661865234375, 70.4964599609375)
```

### PyMuPDF version

1.23.8

### Operating system

Linux

### Python version

3.11

**Comment by richardpaulhudson** [comment]
[with_table.pdf](https://github.com/pymupdf/PyMuPDF/files/13873600/with_table.pdf)
[without_table.pdf](https://github.com/pymupdf/PyMuPDF/files/13873601/without_table.pdf)

**Comment by JorjMcKie** [comment]
Ah, I know the reason:

Within method `find_tables()` we are setting `fitz.TOOLS.set_small_glyph_heights(True)` - a global variable: therefore the permanent, cross document effect.
The intention is to reset it to the previous value, but obviously one return path from the method did not execute this resetting.

The different values pertain to the bbox **_heights_** only. Small glyph heights cut away empty space above and below characters - an effect we need for more precise determination of whether a character belongs to a table cell.

You can use this yourself any time - in fact it is recommended to do so if you need the minimum textbox, e.g. for redactions.

**Comment by richardpaulhudson** [comment]
Thank you very much for the super quick answer! A workaround setting the value back to `False` has indeed solved the problem.

**Comment by JorjMcKie** [comment]
Let's keep the issue open until a corrected version in fact has been published.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.13.
